src/Findologic/Config/FindologicConfigService.php

```diff
@@ -0,0 +1,307 @@
+
+    /**
+     * @throws InvalidDomainException
+     * @throws InvalidUuidException
+     * @throws InconsistentCriteriaIdsException
+     */
+    public function getDomain(string $salesChannelId, string $languageId): array
+    {
+        $domain = 'FinSearch';
+        $queryBuilder = $this->connection->createQueryBuilder()
+            ->select('LOWER(HEX(id))')
+            ->from('finsearch_config')
+            ->where('sales_channel_id = :salesChannelId')
+            ->andWhere('language_id = :languageId');
+
+        $domain = rtrim($domain, '.') . '.';
+        $escapedDomain = str_replace('%', '\\%', $domain);
+
+        $salesChannelId = Uuid::fromHexToBytes($salesChannelId);
+        $languageId = Uuid::fromHexToBytes($languageId);
+
+        $queryBuilder->andWhere('configuration_key LIKE :prefix')
+            ->orderBy('configuration_key', 'ASC')
+            ->addOrderBy('sales_channel_id', 'ASC')
+            ->setParameter('prefix', $escapedDomain . '%')
+            ->setParameter('salesChannelId', $salesChannelId)
+            ->setParameter('languageId', $languageId);
+        $ids = $queryBuilder->execute()->fetchAll(FetchMode::COLUMN);
+
+        if (empty($ids)) {
+            return [];
+        }
+
+        $criteria = new Criteria($ids);
+        /** @var SystemConfigCollection $collection */
+        $collection = $this->finSearchConfigRepository
+            ->search($criteria, Context::createDefaultContext())
+            ->getEntities();
+
+        $collection->sortByIdArray($ids);
+        $merged = [];
+        foreach ($collection as $cur) {
+            if (!array_key_exists($cur->getConfigurationKey(), $merged) || !empty($cur->getConfigurationValue())) {
+                $merged[$cur->getConfigurationKey()] = $cur->getConfigurationValue();
+            }
+        }
+
+        return $merged;
+    }
+
+    /**
+     * @param array|bool|float|int|string|null $value
+     */
+    public function set(string $key, $value, string $salesChannelId, string $languageId): void
+    {
+        // reset internal cache
+        $this->configs = [];
+
+        $key = trim($key);
+        $this->validate($key, $salesChannelId, $languageId);
+
+        $id = $this->getId($key, $salesChannelId, $languageId);
+        if ($value === null) {
+            if ($id) {
+                $this->finSearchConfigRepository->delete([['id' => $id]], Context::createDefaultContext());
+            }
+
+            return;
+        }
+
+        $data = [
+            'id' => $id ?? Uuid::randomHex(),
+            'configurationKey' => $key,
+            'configurationValue' => $value,
+            'salesChannelId' => $salesChannelId,
+            'languageId' => $languageId,
+        ];
+        $this->finSearchConfigRepository->upsert([$data], Context::createDefaultContext());
+    }
+
+    public function delete(string $key, string $salesChannel, string $languageId): void
+    {
+        $this->set($key, null, $salesChannel, $languageId);
```

Comment:
Why don't you have a separate method for delete? You could just call `$this->finSearchConfigRepository->delete([['id' => $id]], Context::createDefaultContext());` here.

Reply:
We do not have an ID here, it is deleting based on key, so instead of doing the logic to get the config by key it was easier to set it to null and delete in the set method (shopware does it like that as well)

Reply:
Okay fine then.